subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/Workarounds.kt

```diff
@@ -31,7 +31,21 @@ object Workarounds {
     fun isIgnoredBeanField(field: Field) =
         ignoredBeanFields.contains(field.name to field.declaringClass.name)
 
+    fun canReadEnvironmentVariable(from: String): Boolean {
+        return isBuildScanPlugin(from) && !shouldDisableInputWorkarounds()
+    }
+
     fun canReadSystemProperty(from: String): Boolean {
+        return isBuildScanPlugin(from) && !shouldDisableInputWorkarounds()
+    }
+
+    private
+    fun isBuildScanPlugin(from: String): Boolean {
         return from.startsWith("com.gradle.scan.plugin.internal.")
     }
+
+    private
+    fun shouldDisableInputWorkarounds(): Boolean {
+        return System.getProperty("org.gradle.internal.disable.input.workarounds")?.equals("true", ignoreCase = true) ?: false
```

Comment:
:question: Won't this trip over the instrumentation and be reported as an input?

Reply:
I don't think we instrument classes in `configuration-cache`, at least that is what my experiments with debugger show.